tests/unit/test_rpm_host_install.py

```diff
@@ -0,0 +1,125 @@
+            # Test for the first case
+            test_old_installed_file = tempfile.mkstemp(prefix=tmpdir)
+            test_old_installed_file_path = test_old_installed_file[1]
+            changed_file_dict = {test_old_installed_file_path : "0"}
+            non_changed_file_dict = {test_old_installed_file_path : "15"}
+
+            RPMHostInstall.rm_add_files_to_host(changed_file_dict, None)
+            self.assertTrue(os.path.exists(test_old_installed_file_path))
+            RPMHostInstall.rm_add_files_to_host(non_changed_file_dict, None)
+            self.assertFalse(os.path.exists(test_old_installed_file_path))
+
+        def setup_and_collect_export_hostfs_files(tmpdir):
+            exports_hostfs_dir = os.path.join(tmpdir, "hostfs")
+            exports_hostfs_tempdir = exports_hostfs_dir +  tmpdir
+            os.makedirs(exports_hostfs_tempdir)
+            # Note about the location here, as we have to make files start with /exports/hostfs
+            # and we want to use the benefit from tmpdir, we have to have the format of /exports/hostfs/tmpdir
+            exports_hostfs_file = os.path.join(exports_hostfs_tempdir, "test.file")
+
+            return exports_hostfs_file, exports_hostfs_dir
+
+        def get_real_exports_file_path(exports_hostfs_file, exports_hostfs_dir):
+            # Find the relative path of the file to exports/hostfs (which is actually the true path under '/' in this contenxt)
+            rel_file_path = os.path.relpath(exports_hostfs_file, exports_hostfs_dir)
+            real_file_path = os.path.join("/", rel_file_path)
+            template_set = [real_file_path]
+            return template_set
+
+        def write_exports_file(exports_hostfs_file):
+            # When the files are already set up, we begin calling function
+            content = {"test_one" : "$test_swap"}
+            with open(exports_hostfs_file, 'w') as json_file:
+                json_file.write(json.dumps(content, indent=4))
+                json_file.write("\n")
+
+        def prepare_test_case(tmpdir):
+            exports_host_file, exports_hostfs_dir = setup_and_collect_export_hostfs_files(tmpdir)
+            template_set = get_real_exports_file_path(exports_host_file, exports_hostfs_dir)
+            write_exports_file(exports_host_file)
+            values = {"test_swap" : "test_result"}
+
+            return template_set, values
+
+        def check_value_swap(template_set, values, tmpdir):
+            real_file_path = template_set[0]
+            RPMHostInstall.rm_add_files_to_host(None, tmpdir, files_template=template_set, values=values)
+            self.assertTrue(os.path.exists(real_file_path))
+            with open(real_file_path, "r") as f:
+                json_val = json.loads(f.read())
+            self.assertEqual(json_val["test_one"], "test_result")
+
+            # Do the cleanup and check last case
+            os.remove(real_file_path)
+
+        try:
+            tmpdir = tempfile.mkdtemp()
```

Comment:
/me thinking. How does python handle exceptions from a subroutine? The reason why I asked is that, in java, when a function generates an exception, it can have 3 options: 1: throw the exception to a 'higher level'  2: handle the exception inside the function 3: let it error out. 

I am concerned a bit as I am not entirely sure how the exception are handled in python. Any clarification will be appreciated, thanks =).